File: src/services/Raggle.Server.WebFront/src/models/DataSource.ts
export type SourceType = "file" | "openapi" | "sqlserver" | "mongo"

export interface DataSource {
  type: SourceType,
  id: string,
  userID: string
  name?: string,
  description?: string,
  details?: any,
  createdAt?: Date,
  updatedAt?: Date,
}

export interface FileDetails {
  files?: FileMeta[]
}

export interface DatabaseDetails {
  connectionString?: string
}

export interface OpenApiDetails {
  schema?: string
}

export interface FileMeta {
  type: string,
  name: string,
  size: number
}

export type SourceDisplay = {
  type: SourceType;
  label: string;
  img: string;
}

export const sourcesList: SourceDisplay[] = [
  { type: "file", label: "File", img: "" },
  { type: "openapi", label: "Open API", img: "" },
  { type: "sqlserver", label: "Microsoft SQL Server", img: "" },
  { type: "mongo", label: "Mongo DB", img: "" },
]

File: src/services/Raggle.Server.Web/Models/DataSource.cs
using System.Text.Json;

namespace Raggle.Server.API.Models;

public class DataSource
{
    public required Guid ID { get; set; }
    public required Guid UserID { get; set; }
    
    public required string Name { get; set; }
    public string? Description { get; set; }

    public required string Type { get; set; }
    public JsonElement? Details { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}

public class FileDetails
{
    public IEnumerable<FileMeta> Files { get; set; } = [];
}

public class DatabaseDetails
{
    public string ConnectionString { get; set; } = string.Empty;
}

public class OpenApiDetails
{
    public string Schema { get; set; } = string.Empty;
}

public class FileMeta
{
    public required string Type { get; set; }
    public required string Name { get; set; }
    public long Size { get; set; }
}
